### PR TITLE
Remove arguments

### LINKED ISSUE #623: Review Deprecated Functions
### Summary

Let's review the functions we deprecated in the previous release and decide whether to hard deprecate them (`lifecycle::deprecate_stop`) or remove them from the codebase entirely.

| Function | Action |
| --- | --- |
| [init_filtered_data code and check argument](https://github.com/insightsengineering/teal.slice/blob/aa05f7feb26bb32fbb5f069ec3e9e77d17450022/R/FilteredData-utils.R#L20-L31) | Remove |

### PR DESCRIPTION
**Pull request by llrs-roche**
# Pull Request

Fixes #623 

After 1 year with an error and with `verify()` in place, it makes sense to remove them.
I didn't keep any documentation or redirect users to other functions/examples, (but it is hard for me to guess if users might not know how to update their code).

The checks will fail due to #629

### CODE REVIEW

**Comment by llrs-roche** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by gogonzo — COMMENTED** [review]
@llrs-roche this needs a NEWS entry

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ### Enhancements
 
 * Reduced the space of the filter panel by not displaying the "add filters" UI in a separate panel.
+* The deprecated parameters `code` and `check` from `init_filtered_data()` were removed.
```
Comment (m7pr) [inline NEWS.md:6]:
```suggestion
* The deprecated parameters `code` and `check` were removed from `init_filtered_data()`.
```

**Comment by m7pr** [comment]
Hey, I left a comment in the main issue https://github.com/insightsengineering/teal.slice/issues/623#issuecomment-2472783188

File: .github/workflows/check.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
     uses: insightsengineering/r.pkg.template/.github/workflows/audit.yaml@main
   r-cmd:
     name: R CMD Check 🧬
-    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
+    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@dependency_resolution@main
```
Comment (m7pr) [inline .github/workflows/check.yaml:24]:
```suggestion
    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
```

**Review by m7pr — APPROVED** [review]
@llrs-roche left you two comments, once they are accepted (the ones about workflows) you are free to merge

**Pull request merged**